include/lbann/data_readers/sample_list_jag.hpp

```diff
@@ -12,10 +12,18 @@
 #include <mpi.h>
 #endif
 
+#include "lbann/utils/file_utils.hpp"
 #include <cereal/types/unordered_map.hpp>
+#include <cereal/types/deque.hpp>
 #include <cereal/types/vector.hpp>
+#include <cereal/types/tuple.hpp>
 #include <cereal/types/string.hpp>
 #include <cereal/types/utility.hpp>
+#include "conduit/conduit_relay_io_hdf5.hpp"
+
+/// Number of system and other files that may be open during execution
+#define LBANN_MAX_OPEN_FILE_MARGIN 128
+#define LBANN_MAX_OPEN_FILE_RETRY 3
```

Comment:
possibly have these as the defaults for two variables that could be overridden on the cmd line